lapce-core/src/command.rs

```diff
@@ -298,6 +298,11 @@ pub enum FocusCommand {
     Rename,
     #[strum(serialize = "confirm_rename")]
     ConfirmRename,
+    /// Terminal selection commands
+    #[strum(serialize = "terminal_copy_selection")]
+    TerminalCopySelection,
```

Comment:
Any reason you don't want to reuse ClipboardCopy/ClipboardPaste?

Reply:
`ClipboardCopy` uses Ctrl+C which the terminal uses to interrupt the current task. `TerminalCopySelection` uses Ctrl+Shift+C which is the shortcut that basically any terminal emulator uses to copy the selection, along with Ctrl+Shift+V (`TerminalPasteSelection`) to paste for consistency.  
I hope my explanation makes sense, English isn't my native language.

Reply:
Got you.

In that case, I would suggest to do this in the keymaps-nonmac.toml

```
[[keymaps]]
key = "ctrl+shift+c"
command = "clipboard_copy"
mode = "t"

[[keymaps]]
key = "ctrl+shift+v"
command = "clipboard_paste"
mode = "t"
```

And you don't need to add anything for keymaps-mac.toml because ```meta+c``` ```meta+v``` would just work on mac.

Reply:
Just tried that, but it doesn't work for some reason. The corresponding match case at `lapce-data/src/terminal.rs:356` is not executed. Any ideas?  

> And you don't need to add anything for keymaps-mac.toml because meta+c meta+v would just work on mac.  

I'm pretty sure the terminal on macOS still uses meta+shift+c and meta+shift+v but don't quote me on that.

Reply:
I found the problem! Apparently, it's the `mode = "t"` line that causes the problem. Don't know what `mode = "t"` does, though.

Reply:
That's because Lapce can be used as a modal editor.

There are a few modes in Lapce, and mode="t" means that key only takes effect when you are in "terminal mode".

Reply:
Okay, I found the problem. In `lapce-data/src/keypress/loader.rs` at line 104, there's a check which excludes all the mappings which are not active in insert mode (unless modal editing is enabled).  
Adding an extra condition to check if the mapping is also **not** active in terminal mode fixes the problem, however I don't know if this would break the intended behavior.

Reply:
I think it should check terminal mode as well.

Reply:
Alright, I'll commit the changes with the extra check for now.